libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -429,14 +426,69 @@ public boolean checkBirthDateCorrelationInCredentials(String userId)
                 <= 1;
     }
 
+    // This method checks the birthdate requirement, which is in violation of GPG45 rules.
+    // However, considering that BAV and Address CRI  does not contain a birthdate,
+    // we add this special handling only for BAV and Address CRI. For other CRIs, we continue to
+    // validate this
+    // requirement.
+    private List<IdentityClaim> getIdentityClaimsForBirthDateCorrelation(
+            List<VcStoreItem> vcStoreItems) throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            IdentityClaim identityClaim =
+                    getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), true);
+            if (isBirthDateEmpty(identityClaim.getBirthDate())) {
+                // excluded BAV and Address CRI
+                if (EXCLUDE_CRIS_TYPES_FOR_DOB_CORRELATION.contains(item.getCredentialIssuer())) {
+                    continue;
+                }
+                addLogMessage(item, "Birthdate property is missing from VC");
+            }
+            identityClaims.add(identityClaim);
+        }
+        return identityClaims;
+    }
+
+    private boolean isBirthDateEmpty(List<BirthDate> birtDates) {
+        return CollectionUtils.isEmpty(birtDates)
+                || birtDates.stream().map(BirthDate::getValue).allMatch(StringUtils::isEmpty);
+    }
+
     public boolean checkNameAndFamilyNameCorrelationInCredentials(String userId)
             throws HttpResponseExceptionWithErrorBody, CredentialParseException {
         final List<VcStoreItem> successfulVCStoreItems =
                 getSuccessfulVCStoreItems(getVcStoreItems(userId));
-        List<IdentityClaim> identityClaims = getIdentityClaims(successfulVCStoreItems);
+        List<IdentityClaim> identityClaims =
+                getIdentityClaimsForNameCorrelation(successfulVCStoreItems);
         return checkNamesForCorrelation(getFullNamesFromCredentials(identityClaims));
     }
 
+    private List<IdentityClaim> getIdentityClaimsForNameCorrelation(List<VcStoreItem> vcStoreItems)
+            throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            IdentityClaim identityClaim =
+                    getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), true);
+            if (isNamesEmpty(identityClaim.getName())) {
+                // excluded Address CRI. Because Address CRI does not contain a name,
+                if (ADDRESS_CRI.equals(item.getCredentialIssuer())) {
```

Comment:
Can we still keep this in a variable alongside the DOB list - even though it's only one item, it makes it easier to see than being hardcoded inside this method :)